### ISSUE TITLE
How to monitor network bandwidth utilization of nodes while training the model?

### ISSUE THREAD

**Issue by orwa-te** [horovod/horovod#2650]
**Environment:**
1. Framework: (TensorFlow)
2. Framework version: 2.3.0
3. Horovod version: 0.20.3
4. MPI version: 4.0.5
5. CUDA version: N/A
6. NCCL version: N/A
7. Python version: 3.7.7
8. OS and version: Centos 7
9. GCC version: 4.8.5

**Checklist:**
1. Did you search issues to find if somebody asked this question before?
2. If your question is about hang, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/running.rst)?
3. If your question is about docker, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/docker.rst)?
4. Did you check if you question is answered in the [troubleshooting guide](https://github.com/horovod/horovod/blob/master/docs/troubleshooting.rst)?

**Your question:**
Is there a way to monitor the network utilization of nodes while communicating with each other to transfer the gradients in order to update the model? I want to measure the size of data sent from one node to another one for a single batch and all batches as well. I think that Tensorboard does not support such a feature

**Comment by tgaddair** [comment]
Hey @orwa-te, you could use something like [iftop](https://www.ex-parrot.com/pdw/iftop/) for this.  Would this work for what you have in mind?

**Comment by orwa-te** [comment]
Hello @tgaddair , `iftop` can be used to monitor all bandwidth utilization on a network interface (from all processes). I need to precisely compute how much data is sent from one node to another during the model parameters transmission in Ring-all-reduce, In other words, what is the size of the data is sent during the `2*(N-1)` transmissions?
I tried to use `nethogs` tool as it can monitor the network utilization for each individual process instead of monitoring the network interface but it is also not sufficient for my case.

**Comment by tgaddair** [comment]
Hey @orwa-te, one approach would be to instrument the NCCL allreduce calls in Horovod similar to how they do it in their internal benchmark suite [here](https://github.com/NVIDIA/nccl-tests).  For example, the size of the buffer passed to ncclAllreduce could be added as a field in the timeline json file when using the timeline feature.  Would that give you the information you're looking for?

**Comment by orwa-te** [comment]
It looks like that the size of the buffer is OK for my case, thanks a lot @tgaddair

**Comment by orwa-te** [comment]
Hello @tgaddair , 
Could you please tell me where exactly I can print the buffer size passed to AllReduce considering using Spark with nodes that have only CPUs?
I am executing my code which is similar to [Spark Rossmann](https://github.com/horovod/horovod/blob/master/examples/spark/keras/keras_spark_rossmann_run.py)

**Comment by tgaddair** [comment]
Hey @orwa-te, for allreduce using MPI you can check the buffer passed in here: https://github.com/horovod/horovod/blob/master/horovod/common/ops/mpi_operations.cc#L56
